### DISCUSSION TITLE
How to define the downloaded PDF Filename

### DISCUSSION THREAD

**Discussion by Stu-Hay** [Hopding/pdf-lib#1164] [Q&A]
Hi,

Apologies if this is a silly question, but I can't seem to find a way to control the filename of the generated PDF. 

I have been able to set the Title Metadata... 

pdfDoc.setTitle('Test', { showInWindowTitleBar: true })

but the downloaded file is still named Unknown.PDF.

Any help with this would be much appreciated.

Regards

Stu

**Comment by Stu-Hay** [comment]
My best guess would be I need to serialise the PDF via pdfDoc.save() and then write some additional code to take those bytes and write a file?

**Reply by bcholmes** [reply]
The way I do it, in-browser, uses an additional library called [download.js](https://github.com/rndme/download) -- this is based on example code from pdf-lib. Use that to "download" the pdf to the user's downloads directory. One of the parameters lets you set an name:

    const pdfBytes = await pdfDoc.save();

    // Trigger the browser to download the PDF document
    download(pdfBytes, "my-file-name.pdf", "application/pdf");

That works well for me.

**Reply by Stu-Hay** [reply]
Thanks, I'll have a look at that.

**Reply by yaakovyitzchak** [reply]
Also just make a blob to download it, then set the name with the download attribute of an "a" node
```
var blob=new Blob([bytesFromDocSave])
var url=URL.createObjectURL(blob)
var a=document.createElement("a")
a.href=url
a.download="Fancy name of PDF.pdf"
a .click()

**Comment by dan-willett** [comment]
@bcholmes The link to download.js you included was the wrong library...  I believe the link you were looking for is: https://github.com/rndme/download
